File: Fortran/0166/0166_0014.f90
       interface operator(.oper.)
       function sub1(i,j)
       integer :: sub1
       real :: i
       intent(in) :: i
       integer :: j
       intent(in) :: j
       end function
       end interface

      complex(kind=4):: a
      character :: cc
      integer :: p ,sub
      cc='a'
      p = sub(a%re,1)
      if (a%re.ne.1.0) print *,'err'
      p = (a%re).oper.(1)
      if (p.ne.1.0) print *,'err'
      print *,'pass'
      end
      function sub(i,j)
      real(kind=4) :: i
      intent(inout) :: i
      integer :: j,sub
      intent(in) :: j
      i=1.0
      sub = i + 2
      end function

       function sub1(i,j)
       integer :: sub1
       real :: i
       intent(in) :: i
       integer :: j
       intent(in) :: j
        sub1=i
       end function

File: C/0166/0166_0014.c
#include <stdio.h>

void sub(int n, int a[], int b[])
{
  int i;
  for (i = 0; i < n; ++i) a[i+1] = a[i] + b[i];
}
int main()
{
  int a[5] = {0,1,2,3,4};
  int b[5] = {5,6,7,8,9};
  sub(4,a,b);
  if (a[0] == 0 && a[1] == 5 && a[2] == 11 && a[3] == 18 && a[4] == 26) {
    puts("OK");
  } else {
    puts("NG");
    printf("%d,%d,%d,%d,%d\n",a[0],a[1],a[2],a[3],a[4]);
  }
}
